nav2_system_tests/src/costmap_filters/test_speed_launch.py

```diff
@@ -55,29 +56,36 @@ def generate_launch_description():
         convert_types=True)
     context = LaunchContext()
     new_yaml = configured_params.perform(context)
+    urdf = os.getenv('TEST_URDF')
+    with open(urdf, 'r') as infp:
+        robot_description = infp.read()
     return LaunchDescription([
         SetEnvironmentVariable('RCUTILS_LOGGING_BUFFERED_STREAM', '1'),
         SetEnvironmentVariable('RCUTILS_LOGGING_USE_STDOUT', '1'),
 
         # Launch gazebo server for simulation
         ExecuteProcess(
-            cmd=['gzserver', '-s', 'libgazebo_ros_init.so',
-                 '--minimal_comms', world],
+            cmd=['gzserver', '-s', 'libgazebo_ros_init.so', '-s', 'libgazebo_ros_factory.so',
+                 world],
+            cwd=[aws_dir],
             output='screen'),
 
-        # TODO(orduno) Launch the robot state publisher instead
-        #              using a local copy of TB3 urdf file
         Node(
-            package='tf2_ros',
-            executable='static_transform_publisher',
+            package='robot_state_publisher',
+            executable='robot_state_publisher',
+            name='robot_state_publisher',
             output='screen',
-            arguments=['0', '0', '0', '0', '0', '0', 'base_footprint', 'base_link']),
+            parameters=[
+                {'use_sim_time': True, 'robot_description': robot_description}
+            ]),
 
         Node(
-            package='tf2_ros',
-            executable='static_transform_publisher',
+            package='gazebo_ros',
```

Comment:
Have you done extensive testing using the gazebo proper model rather than faking it out? That's a big change to add the model / URDF, if it works, that's great! But would have been probably good to include in a separate PR so we don't have one massive one with subtle changes.

Have you run CI a bunch to make sure this doesn't add flaky tests and works reliably? I know it fails in debug mode on occasion, but if compiling as release, this test should be relatively solid. 

Ditto with all others that match this change

Reply:
I've done testing only locally on my laptop. Never had the chance to try on CI.

Reply:
OK, we can do some more testing in CI as the releases turn over. I'll probably also try this myself and attach a gazebo frontend session so I can watch just to make sure its reasonable replacements. But assuming they are, and they turn over in CI fine, this is OK with me